File: ruby/lib/ccpayment/services/user_swap_service.rb
module CCPayment
  module Services
    class UserSwapService
      def initialize(client); @client = client; end
      def get_user_swap_record(data); @client.post('/getUserSwapRecord', data); end
      def get_user_swap_record_list(data = {}); @client.post('/getUserSwapRecordList', data); end
      def user_swap(data); @client.post('/userSwap', data); end
    end
  end
end

File: java/src/main/java/com/ccpayment/services/UserSwapService.java
package com.ccpayment.services;

import com.ccpayment.APIError;
import com.ccpayment.Client;
import com.google.gson.JsonObject;

import java.io.IOException;

public class UserSwapService {
    private final Client client;
    public UserSwapService(Client client) { this.client = client; }
    public JsonObject getUserSwapRecord(JsonObject data) throws APIError, IOException { return client.post("/getUserSwapRecord", data, JsonObject.class); }
    public JsonObject getUserSwapRecordList(JsonObject data) throws APIError, IOException { return client.post("/getUserSwapRecordList", data, JsonObject.class); }
    public JsonObject userSwap(JsonObject data) throws APIError, IOException { return client.post("/userSwap", data, JsonObject.class); }
}
